File: schemaorggo/airline.go
package schemaorggo

import "encoding/json"

// Airline see : https://schema.org/Airline
type Airline struct {
	Organization

	typeContext

	// BoardingPolicy see : https://schema.org/boardingPolicy
	// The type of boarding policy used by the airline (e.g. zone-based or group-based).
	// types : BoardingPolicyType
	BoardingPolicy []*BoardingPolicyType `json:"boardingPolicy,omitempty"`

	// IataCode see : https://schema.org/iataCode
	// IATA identifier for an airline or airport.
	// types : Text
	IataCode []string `json:"iataCode,omitempty"`
}

func (v Airline) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.Organization.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.BoardingPolicy
		if len(v.BoardingPolicy) == 1 {
			value = v.BoardingPolicy[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["boardingPolicy"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.IataCode
		if len(v.IataCode) == 1 {
			value = v.IataCode[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["iataCode"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v Airline) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "Airline"

	return data, nil
}

func (v Airline) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/airline.php
<?php

class Airline extends Organization implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'Airline';
	
	/**
	 * The type of boarding policy used by the airline (e.g. zone-based or group-based).
	 * see : https://schema.org/boardingPolicy
	 * @var \BoardingPolicyType|\BoardingPolicyType[]
	 */
	public var $boarding_policy;
	
	/**
	 * IATA identifier for an airline or airport.
	 * see : https://schema.org/iataCode
	 * @var string|string[]
	 */
	public var $iata_code;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'Airline'
		);
		
		$serialized = so_json_serialize( $this->boarding_policy );
		if ( ! empty( $serialized ) ) {
			$out['boardingPolicy'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->iata_code );
		if ( ! empty( $serialized ) ) {
			$out['iataCode'] = $serialized;
		}
		
		return $out;
	}
}
